src/spikeinterface/widgets/rasters.py

```diff
@@ -25,64 +293,64 @@ class RasterWidget(BaseWidget):
     """
 
     def __init__(
-        self, sorting, segment_index=None, unit_ids=None, time_range=None, color="k", backend=None, **backend_kwargs
+        self,
+        sorting=None,
+        sorting_analyzer=None,
+        segment_index=None,
+        unit_ids=None,
+        time_range=None,
+        color="k",
+        backend=None,
+        **backend_kwargs,
     ):
+        if sorting is None and sorting_analyzer is None:
+            raise Exception("Must supply either a sorting or a sorting_analyzer")
+        if sorting is not None and sorting_analyzer is not None:
+            raise Exception("Should supply either a sorting or a sorting_analyzer, not both")
+        if sorting_analyzer is not None:
+            sorting = sorting_analyzer.sorting
+
         sorting = self.ensure_sorting(sorting)
 
         if segment_index is None:
             if sorting.get_num_segments() != 1:
                 raise ValueError("You must provide segment_index=...")
             segment_index = 0
 
-        if time_range is None:
-            frame_range = [0, sorting.to_spike_vector()[-1]["sample_index"]]
-            time_range = [f / sorting.sampling_frequency for f in frame_range]
-        else:
+        if unit_ids is None:
+            unit_ids = sorting.unit_ids
+
+        all_spiketrains = {
+            unit_id: sorting.get_unit_spike_train(unit_id, segment_index=0, return_times=True) for unit_id in unit_ids
```

Comment:
```suggestion
            unit_id: sorting.get_unit_spike_train(unit_id, segment_index=segment_index, return_times=True) for unit_id in unit_ids
```
should still be user-specifiable. But I think we need to fix the logic above to make sure nothing bad sneaks through.